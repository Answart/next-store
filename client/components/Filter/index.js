import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Router from 'next/router';
import { StyledFilter } from '../styles/FilterStyles';
import FilterSection from './FilterSection';
import FilterList from './FilterList';
import FilterRange from './FilterRange';
import { DEPARTMENTS, CATEGORIES, COLORS, SIZES } from '../../config';
import { capWord } from '../../lib/utils';


class Filter extends Component {
  static propTypes = {
    pageQuery: PropTypes.object.isRequired,
    products: PropTypes.arrayOf(PropTypes.object)
  };
  state = {
    showCategory: true,
    showColor: true,
    showSize: true,
    showPrice: true,
    showBrand: true
  };
  toggleFilter = e => {
    if (!!e && e.preventDefault) e.preventDefault();
    const { name } = e.currentTarget;

    this.setState(state => ({ [name]: !state[name] }));
  };
  updateFilter = e => {
    if (!!e && e.preventDefault) e.preventDefault();
    const { name, value } = e.currentTarget;
    if (!name) return;
    const { pageQuery } = this.props;

    if (!!value && pageQuery[name] !== value) {
      pageQuery[name] = value;
    } else {
      delete pageQuery[name];
    }

    Router.push({
      pathname: "/shop",
      query: { ...pageQuery }
    });
  };
  render() {
    const { pageQuery } = this.props;
    const {
      showCategory, showColor, showSize, showPrice, showBrand
    } = this.state;
    const categoryListType = pageQuery.department
      ? 'category'
      : 'department';
    const categoryList = categoryListType === 'category'
      ? CATEGORIES[pageQuery.department]
      : DEPARTMENTS;
    return (
      <StyledFilter data-test="filter">
        <FilterSection
          showName='showCategory'
          name={categoryListType}
          currentFilter={pageQuery[categoryListType]}
          showFilter={showCategory}
          toggleFilter={this.toggleFilter}
          updateFilter={this.updateFilter}
        >
          {showCategory && (
            <FilterList
              name={categoryListType}
              list={categoryList}
              currentFilter={pageQuery[categoryListType]}
              updateFilter={this.updateFilter}
            />
          )}
        </FilterSection>

        <FilterSection
          showName='showColor'
          name='color'
          circleHover={true}
          currentFilter={pageQuery['color']}
          showFilter={showColor}
          toggleFilter={this.toggleFilter}
          updateFilter={this.updateFilter}
        >
          {showColor && (
            <FilterList
              name='color'
              circleHover={true}
              list={COLORS}
              currentFilter={pageQuery['color']}
              updateFilter={this.updateFilter}
            />
          )}
        </FilterSection>

        <FilterSection
          showName='showSize'
          name='size'
          circleHover={true}
          currentFilter={pageQuery['size']}
          showFilter={showSize}
          toggleFilter={this.toggleFilter}
          updateFilter={this.updateFilter}
        >
          {showSize && (
            <FilterList
              name='size'
              circleHover={true}
              list={SIZES}
              currentFilter={pageQuery['size']}
              updateFilter={this.updateFilter}
            />
          )}
        </FilterSection>

        <FilterSection
          showName='showPrice'
          name='price'
          circleHover={true}
          currentFilter={pageQuery['price']}
          showFilter={showPrice}
          toggleFilter={this.toggleFilter}
          updateFilter={this.updateFilter}
        >
          {showPrice && (
            <FilterRange
              price_gte={pageQuery['price_gte']}
              price_lte={pageQuery['price_lte']}
              updateFilter={this.updateFilter}
            />
          )}
        </FilterSection>

        <FilterSection
          showName='showBrand'
          name='brand'
          circleHover={true}
          currentFilter={pageQuery['brand']}
          showFilter={showBrand}
          toggleFilter={this.toggleFilter}
          updateFilter={this.updateFilter}
        >
          {showBrand && (
            <div className={`filter-show filter-show-brand`}>
              {['Converse', 'brand2'].map((listItem, i) => (
                <div key={`brand-${listItem}`} className="filter-brand">
                  <label htmlFor={`brand-${listItem}`}>
                    <input id={`brand-${listItem}`}
                      type="radio"
                      name="brand"
                      key={name}
                      value={listItem}
                      onChange={this.updateFilter}
                      checked={!!pageQuery['brand'] && pageQuery['brand'] == listItem}
                    /><span className='brand-label'>{capWord(listItem)}</span>
                  </label>
                </div>
              ))}
            </div>
          )}
        </FilterSection>
      </StyledFilter>
    );
  }
};


export default Filter;
