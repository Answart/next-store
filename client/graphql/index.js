export {
  TOGGLE_LOCAL_CARTOPEN_MUTATION,
  SIGNUP_MUTATION,
  SIGNIN_MUTATION,
  SIGNOUT_MUTATION,
  REQUEST_PASSWORD_RESET_MUTATION,
  RESET_PASSWORD_MUTATION,
  UPDATE_PERMISSIONS_MUTATION,
  CREATE_IMAGE_MUTATION,
  CREATE_PRODUCT_MUTATION,
  UPDATE_PRODUCT_MUTATION,
  DELETE_PRODUCT_MUTATION,
  CREATE_PROD_VARIANT_MUTATION,
  UPDATE_PROD_VARIANT_MUTATION,
  DELETE_PROD_VARIANT_MUTATION,
  ADD_TO_CART_MUTATION,
  UPDATE_CARTITEM_MUTATION,
  REMOVE_FROM_CART_MUTATION,
  CREATE_ORDER_MUTATION,
} from './Mutation';
export {
  LOCAL_CARTOPEN_QUERY,
  LOCAL_USER_QUERY,
  ALL_USERS_QUERY,
  CURRENT_USER_QUERY,
  ORDER_QUERY,
  ORDERS_QUERY,
  ORDER_ITEMS_QUERY,
  PRODUCT_QUERY,
  SHOP_PRODUCTS_QUERY,
  PAGINATION_QUERY,
} from './Query';
