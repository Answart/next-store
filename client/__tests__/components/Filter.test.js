import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import wait from 'waait';
import { MockedProvider } from 'react-apollo/test-utils';
import { ApolloConsumer } from 'react-apollo';
import Router from 'next/router';
import Filter from '../../components/Filter';
import { mockProduct, shopProductsQueryFilterMock } from '../../lib/test-utils/mocks';
import { SHOP_PRODUCTS_QUERY } from '../../graphql';

const mocks = [
  shopProductsQueryFilterMock({ variables: { department: 'accessories' }, products: true }),
  shopProductsQueryFilterMock({ variables: { department: 'accessories', category: 'jewelry' }, products: true }),
  shopProductsQueryFilterMock({ variables: { department: 'accessories', category: 'bags' }, products: false }),
  shopProductsQueryFilterMock({ variables: { department: 'accessories' }, products: true }),
];
Router.onRouteChangeStart = jest.fn();
Router.onRouteChangeComplete = jest.fn();
Router.onRouteChangeError = jest.fn();
Router.router = {
  push: jest.fn(),
  replace: jest.fn(),
  go: jest.fn(),
  createHref: jest.fn(),
  createLocation: jest.fn(),
  isActive: jest.fn(),
  matcher: {
    match: jest.fn(),
    getRoutes: jest.fn(),
    isActive: jest.fn(),
    format: jest.fn()
  },
  addTransitionHook: jest.fn(),
  preventDefault: () => {}
};


describe('<Filter />', () => {
  let wrapper, apolloClient;
  beforeAll(() => {
    wrapper = mount(
      <MockedProvider mocks={mocks}>
        <ApolloConsumer>
          {client => {
            apolloClient = client;
            return (
              <Filter
                pageQuery={{ department: 'accessories' }}
                products={[mockProduct]}
              />
            )
          }}
        </ApolloConsumer>
      </MockedProvider>
    );
  });
  afterAll(() => {
    wrapper.unmount();
    jest.clearAllMocks();
  });

  it('matches the snap shot', async () => {
    await wait();
    wrapper.update();
    expect(toJSON(wrapper.find('Filter'))).toMatchSnapshot();
  });

  it('renders properly', async () => {
    expect(wrapper.find('#filter-department').length).toBe(0);
    expect(wrapper.find('#filter-category').length).toBe(1);
    expect(wrapper.find('#filter-color').length).toBe(1);
    expect(wrapper.find('#filter-size').length).toBe(1);
    expect(wrapper.find('#filter-price').length).toBe(1);
    expect(wrapper.find('#filter-brand').length).toBe(1);
  });

  describe('Filter Section', async () => {
    let categFilter;
    beforeAll(() => categFilter = wrapper.find('#filter-category'));

    it('initial banner renders properly', async () => {
      const labelBtn = categFilter.find('button#filter-category-label');
      expect(labelBtn.length).toBe(1);
      expect(labelBtn.text()).toBe('CATEGORY');
      const clearBtn = categFilter.find('button#filter-clear-category-btn');
      expect(clearBtn.length).toBe(1);
      expect(clearBtn.text()).toBe('');
      const arrowBtn = categFilter.find('button#filter-show-category-btn');
      expect(arrowBtn.length).toBe(1);
      expect(arrowBtn.find('SvgIcon').props().name).toBe('downArrow');
    });

    it('clicking a filter option call properly', async () => {
      const categFilter = wrapper.find('#filter-category');
      expect(categFilter.find('button#category-jewelry').props().disabled).toBe(false);
      expect(wrapper.find('Filter').props().pageQuery.category).toBe(undefined);
      categFilter.find('button#category-jewelry').simulate('click');
      await wait();
      wrapper.find('Filter').instance().forceUpdate();
      expect(Router.router.push).toHaveBeenCalled();
      expect(Router.router.push).toHaveBeenCalledWith({
        pathname: "/shop",
        query: {
          department: 'accessories', category: 'jewelry'
        }
      });
      const { data: { products } } = await apolloClient.query({
        query: SHOP_PRODUCTS_QUERY,
        variables: {
          department: 'accessories', category: 'jewelry'
        }
      });
      expect(products.length).toBe(1);
      wrapper.update();
      expect(wrapper.find('Filter').props().pageQuery.category).toBe('jewelry');
      expect(wrapper.find('#filter-category').find('button#category-jewelry').props().disabled).toBe(true);
    });

    it('clicking a different filter option call properly', async () => {
      const categFilter = wrapper.find('#filter-category');
      expect(categFilter.find('button#category-jewelry').props().disabled).toBe(true);
      expect(categFilter.find('button#category-bags').props().disabled).toBe(false);
      expect(wrapper.find('Filter').props().pageQuery.category).toBe('jewelry');
      categFilter.find('button#category-bags').simulate('click');
      await wait();
      wrapper.find('Filter').instance().forceUpdate();
      expect(Router.router.push).toHaveBeenCalled();
      expect(Router.router.push).toHaveBeenCalledWith({
        pathname: "/shop",
        query: {
          department: 'accessories', category: 'bags'
        }
      });
      const { data: { products } } = await apolloClient.query({
        query: SHOP_PRODUCTS_QUERY,
        variables: {
          department: 'accessories', category: 'bags'
        }
      });
      expect(products.length).toBe(0);
      wrapper.update();
      expect(wrapper.find('Filter').props().pageQuery.category).toBe('bags');
      expect(wrapper.find('#filter-category').find('button#category-jewelry').props().disabled).toBe(false);
      expect(wrapper.find('#filter-category').find('button#category-bags').props().disabled).toBe(true);
    });

    it("clicking 'Clear' removes category from query properly", async () => {
      const categClearBtn = wrapper.find('button#filter-clear-category-btn');
      expect(wrapper.find('Filter').props().pageQuery.category).toBe('bags');
      expect(categClearBtn.length).toBe(1);
      expect(categClearBtn.props().disabled).toBe(false);
      expect(categClearBtn.text()).toBe('Clear');
      categClearBtn.simulate('click');
      await wait();
      wrapper.find('Filter').instance().forceUpdate();
      expect(Router.router.push).toHaveBeenCalled();
      expect(Router.router.push).toHaveBeenCalledWith({
        pathname: "/shop",
        query: {
          department: 'accessories'
        }
      });
      const { data: { products } } = await apolloClient.query({
        query: SHOP_PRODUCTS_QUERY,
        variables: {
          department: 'accessories'
        }
      });
      expect(products.length).toBe(1);
      wrapper.update();
      expect(wrapper.find('Filter').props().pageQuery.category).toBe(undefined);
      expect(wrapper.find('button#filter-clear-category-btn').props().disabled).toBe(true);
      expect(wrapper.find('button#filter-clear-category-btn').text()).toBe('');
    });

    it("clicking Arrow hides filter section", async () => {
      const categArrowBtn = categFilter.find('button#filter-show-category-btn');
      expect(wrapper.find('Filter').instance().state.showCategory).toBe(true);
      expect(categArrowBtn.length).toBe(1);
      expect(categArrowBtn.find('SvgIcon').props().name).toBe('downArrow');
      categArrowBtn.simulate('click');
      await wait();
      wrapper.update();
      expect(wrapper.find('Filter').instance().state.showCategory).toBe(false);
      expect(wrapper.find('button#filter-show-category-btn').find('SvgIcon').props().name).toBe('upArrow');
    });
  });
});
