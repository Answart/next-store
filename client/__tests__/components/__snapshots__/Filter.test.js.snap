// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> matches the snap shot 1`] = `
<Filter
  pageQuery={
    Object {
      "department": "accessories",
    }
  }
  products={
    Array [
      Object {
        "__typename": "Product",
        "brand": "Peggs",
        "category": "jewelry",
        "department": "accessories",
        "description": "Limited Edition watch from the 2018 Fall fashion line.",
        "id": "pr0duct1d",
        "image": Object {
          "__typename": "Image",
          "cloudinary_id": "111111",
          "height": 22,
          "id": "1m4g31d",
          "image_url": "peggswatch1.jpg",
          "large_image_url": "peggswatch2.jpg",
          "name": "peggswatch1",
          "transformation": "",
          "width": 22,
        },
        "online": false,
        "title": "Peggs Gold Edition Analog Watch Peggs Gold Edition Analog ",
        "user": Object {
          "__typename": "User",
          "cart": Array [],
          "email": "Waldo.Yundt@Mosciski.biz",
          "id": "us3r1d",
          "name": "Ms. Josue McCullough",
          "password": "unique-password",
          "permissions": Array [
            "ADMIN",
          ],
        },
        "variants": Array [],
      },
    ]
  }
>
  <Filter__FilterStyles
    data-test="filter"
  >
    <div
      className="Filter__FilterStyles-sc-129b01r-0 cfKNls"
      data-test="filter"
    >
      <FilterSection
        list={
          Array [
            "jewelry",
            "bags",
            "wallets",
            "belts",
            "sunglasses",
            "makeup",
            "hair",
            "hats",
          ]
        }
        name="category"
        showFilter={true}
        showName="showCategory"
        toggleFilter={[Function]}
        updateFilter={[Function]}
      >
        <div
          className="filter"
          id="filter-category"
        >
          <div
            className="filter-banner"
          >
            <button
              id="filter-category-label"
              name="showCategory"
              onClick={[Function]}
            >
              <h4>
                CATEGORY
              </h4>
            </button>
            <button
              className="filter-clear-btn"
              disabled={true}
              id="filter-clear-category-btn"
              name="category"
              onClick={[Function]}
              value={null}
            />
            <button
              className="filter-show-btn"
              id="filter-show-category-btn"
              name="showCategory"
              onClick={[Function]}
            >
              <SvgIcon
                name="downArrow"
                width={10}
              >
                <svg
                  className=""
                  height="10px"
                  style={Object {}}
                  viewBox="0 0 451.847 451.847"
                  width="10px"
                  xmlns="http://www.w3.org/2000/svg"
                  xmlnsXlink="http://www.w3.org/1999/xlink"
                >
                  <g>
                    <path
                      d="M225.923,354.706c-8.098,0-16.195-3.092-22.369-9.263L9.27,151.157c-12.359-12.359-12.359-32.397,0-44.751 c12.354-12.354,32.388-12.354,44.748,0l171.905,171.915l171.906-171.909c12.359-12.354,32.391-12.354,44.744,0 c12.365,12.354,12.365,32.392,0,44.751L248.292,345.449C242.115,351.621,234.018,354.706,225.923,354.706z"
                      fill="#6d6c6c"
                      key="0"
                    />
                  </g>
                </svg>
              </SvgIcon>
            </button>
          </div>
          <div
            className="filter-show filter-show-category"
          >
            <button
              className="undrln-btn false"
              disabled={false}
              id="category-jewelry"
              key="0"
              name="category"
              onClick={[Function]}
              title="Refine by category: Jewelry"
              value="jewelry"
            >
              Jewelry
            </button>
            <button
              className="undrln-btn false"
              disabled={false}
              id="category-bags"
              key="1"
              name="category"
              onClick={[Function]}
              title="Refine by category: Bags"
              value="bags"
            >
              Bags
            </button>
            <button
              className="undrln-btn false"
              disabled={false}
              id="category-wallets"
              key="2"
              name="category"
              onClick={[Function]}
              title="Refine by category: Wallets"
              value="wallets"
            >
              Wallets
            </button>
            <button
              className="undrln-btn false"
              disabled={false}
              id="category-belts"
              key="3"
              name="category"
              onClick={[Function]}
              title="Refine by category: Belts"
              value="belts"
            >
              Belts
            </button>
            <button
              className="undrln-btn false"
              disabled={false}
              id="category-sunglasses"
              key="4"
              name="category"
              onClick={[Function]}
              title="Refine by category: Sunglasses"
              value="sunglasses"
            >
              Sunglasses
            </button>
            <button
              className="undrln-btn false"
              disabled={false}
              id="category-makeup"
              key="5"
              name="category"
              onClick={[Function]}
              title="Refine by category: Makeup"
              value="makeup"
            >
              Makeup
            </button>
            <button
              className="undrln-btn false"
              disabled={false}
              id="category-hair"
              key="6"
              name="category"
              onClick={[Function]}
              title="Refine by category: Hair"
              value="hair"
            >
              Hair
            </button>
            <button
              className="undrln-btn false"
              disabled={false}
              id="category-hats"
              key="7"
              name="category"
              onClick={[Function]}
              title="Refine by category: Hats"
              value="hats"
            >
              Hats
            </button>
          </div>
        </div>
      </FilterSection>
      <FilterSection
        circleHover={true}
        list={
          Array [
            "red",
            "orange",
            "yellow",
            "green",
            "blue",
            "purple",
            "pink",
            "brown",
            "black",
            "white",
            "silver",
            "gold",
            "multi",
          ]
        }
        name="color"
        showFilter={true}
        showName="showColor"
        toggleFilter={[Function]}
        updateFilter={[Function]}
      >
        <div
          className="filter"
          id="filter-color"
        >
          <div
            className="filter-banner"
          >
            <button
              id="filter-color-label"
              name="showColor"
              onClick={[Function]}
            >
              <h4>
                COLOR
              </h4>
            </button>
            <button
              className="filter-clear-btn"
              disabled={true}
              id="filter-clear-color-btn"
              name="color"
              onClick={[Function]}
              value={null}
            />
            <button
              className="filter-show-btn"
              id="filter-show-color-btn"
              name="showColor"
              onClick={[Function]}
            >
              <SvgIcon
                name="downArrow"
                width={10}
              >
                <svg
                  className=""
                  height="10px"
                  style={Object {}}
                  viewBox="0 0 451.847 451.847"
                  width="10px"
                  xmlns="http://www.w3.org/2000/svg"
                  xmlnsXlink="http://www.w3.org/1999/xlink"
                >
                  <g>
                    <path
                      d="M225.923,354.706c-8.098,0-16.195-3.092-22.369-9.263L9.27,151.157c-12.359-12.359-12.359-32.397,0-44.751 c12.354-12.354,32.388-12.354,44.748,0l171.905,171.915l171.906-171.909c12.359-12.354,32.391-12.354,44.744,0 c12.365,12.354,12.365,32.392,0,44.751L248.292,345.449C242.115,351.621,234.018,354.706,225.923,354.706z"
                      fill="#6d6c6c"
                      key="0"
                    />
                  </g>
                </svg>
              </SvgIcon>
            </button>
          </div>
          <div
            className="filter-show filter-show-color"
          >
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="color-red"
              key="0"
              name="color"
              onClick={[Function]}
              title="Refine by color: Red"
              value="red"
            >
              <div
                className="color-sphere-sample red-color-sample"
              />
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="color-orange"
              key="1"
              name="color"
              onClick={[Function]}
              title="Refine by color: Orange"
              value="orange"
            >
              <div
                className="color-sphere-sample orange-color-sample"
              />
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="color-yellow"
              key="2"
              name="color"
              onClick={[Function]}
              title="Refine by color: Yellow"
              value="yellow"
            >
              <div
                className="color-sphere-sample yellow-color-sample"
              />
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="color-green"
              key="3"
              name="color"
              onClick={[Function]}
              title="Refine by color: Green"
              value="green"
            >
              <div
                className="color-sphere-sample green-color-sample"
              />
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="color-blue"
              key="4"
              name="color"
              onClick={[Function]}
              title="Refine by color: Blue"
              value="blue"
            >
              <div
                className="color-sphere-sample blue-color-sample"
              />
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="color-purple"
              key="5"
              name="color"
              onClick={[Function]}
              title="Refine by color: Purple"
              value="purple"
            >
              <div
                className="color-sphere-sample purple-color-sample"
              />
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="color-pink"
              key="6"
              name="color"
              onClick={[Function]}
              title="Refine by color: Pink"
              value="pink"
            >
              <div
                className="color-sphere-sample pink-color-sample"
              />
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="color-brown"
              key="7"
              name="color"
              onClick={[Function]}
              title="Refine by color: Brown"
              value="brown"
            >
              <div
                className="color-sphere-sample brown-color-sample"
              />
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="color-black"
              key="8"
              name="color"
              onClick={[Function]}
              title="Refine by color: Black"
              value="black"
            >
              <div
                className="color-sphere-sample black-color-sample"
              />
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="color-white"
              key="9"
              name="color"
              onClick={[Function]}
              title="Refine by color: White"
              value="white"
            >
              <div
                className="color-sphere-sample white-color-sample"
              />
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="color-silver"
              key="10"
              name="color"
              onClick={[Function]}
              title="Refine by color: Silver"
              value="silver"
            >
              <div
                className="color-sphere-sample silver-color-sample"
              />
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="color-gold"
              key="11"
              name="color"
              onClick={[Function]}
              title="Refine by color: Gold"
              value="gold"
            >
              <div
                className="color-sphere-sample gold-color-sample"
              />
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="color-multi"
              key="12"
              name="color"
              onClick={[Function]}
              title="Refine by color: Multi"
              value="multi"
            >
              <div
                className="color-sphere-sample multi-color-sample"
              />
            </button>
          </div>
        </div>
      </FilterSection>
      <FilterSection
        circleHover={true}
        list={
          Array [
            "XXS",
            "XS",
            "S",
            "M",
            "L",
            "XL",
            "1X",
            "2X",
            "3X",
            "4X",
            "0",
            "2",
            "4",
            "6",
            "8",
            "10",
            "12",
            "14",
            "16",
            "18",
          ]
        }
        name="size"
        showFilter={true}
        showName="showSize"
        toggleFilter={[Function]}
        updateFilter={[Function]}
      >
        <div
          className="filter"
          id="filter-size"
        >
          <div
            className="filter-banner"
          >
            <button
              id="filter-size-label"
              name="showSize"
              onClick={[Function]}
            >
              <h4>
                SIZE
              </h4>
            </button>
            <button
              className="filter-clear-btn"
              disabled={true}
              id="filter-clear-size-btn"
              name="size"
              onClick={[Function]}
              value={null}
            />
            <button
              className="filter-show-btn"
              id="filter-show-size-btn"
              name="showSize"
              onClick={[Function]}
            >
              <SvgIcon
                name="downArrow"
                width={10}
              >
                <svg
                  className=""
                  height="10px"
                  style={Object {}}
                  viewBox="0 0 451.847 451.847"
                  width="10px"
                  xmlns="http://www.w3.org/2000/svg"
                  xmlnsXlink="http://www.w3.org/1999/xlink"
                >
                  <g>
                    <path
                      d="M225.923,354.706c-8.098,0-16.195-3.092-22.369-9.263L9.27,151.157c-12.359-12.359-12.359-32.397,0-44.751 c12.354-12.354,32.388-12.354,44.748,0l171.905,171.915l171.906-171.909c12.359-12.354,32.391-12.354,44.744,0 c12.365,12.354,12.365,32.392,0,44.751L248.292,345.449C242.115,351.621,234.018,354.706,225.923,354.706z"
                      fill="#6d6c6c"
                      key="0"
                    />
                  </g>
                </svg>
              </SvgIcon>
            </button>
          </div>
          <div
            className="filter-show filter-show-size"
          >
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="size-XXS"
              key="0"
              name="size"
              onClick={[Function]}
              title="Refine by size: XXS"
              value="XXS"
            >
              XXS
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="size-XS"
              key="1"
              name="size"
              onClick={[Function]}
              title="Refine by size: XS"
              value="XS"
            >
              XS
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="size-S"
              key="2"
              name="size"
              onClick={[Function]}
              title="Refine by size: S"
              value="S"
            >
              S
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="size-M"
              key="3"
              name="size"
              onClick={[Function]}
              title="Refine by size: M"
              value="M"
            >
              M
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="size-L"
              key="4"
              name="size"
              onClick={[Function]}
              title="Refine by size: L"
              value="L"
            >
              L
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="size-XL"
              key="5"
              name="size"
              onClick={[Function]}
              title="Refine by size: XL"
              value="XL"
            >
              XL
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="size-1X"
              key="6"
              name="size"
              onClick={[Function]}
              title="Refine by size: 1X"
              value="1X"
            >
              1X
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="size-2X"
              key="7"
              name="size"
              onClick={[Function]}
              title="Refine by size: 2X"
              value="2X"
            >
              2X
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="size-3X"
              key="8"
              name="size"
              onClick={[Function]}
              title="Refine by size: 3X"
              value="3X"
            >
              3X
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="size-4X"
              key="9"
              name="size"
              onClick={[Function]}
              title="Refine by size: 4X"
              value="4X"
            >
              4X
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="size-0"
              key="10"
              name="size"
              onClick={[Function]}
              title="Refine by size: 0"
              value="0"
            >
              0
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="size-2"
              key="11"
              name="size"
              onClick={[Function]}
              title="Refine by size: 2"
              value="2"
            >
              2
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="size-4"
              key="12"
              name="size"
              onClick={[Function]}
              title="Refine by size: 4"
              value="4"
            >
              4
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="size-6"
              key="13"
              name="size"
              onClick={[Function]}
              title="Refine by size: 6"
              value="6"
            >
              6
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="size-8"
              key="14"
              name="size"
              onClick={[Function]}
              title="Refine by size: 8"
              value="8"
            >
              8
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="size-10"
              key="15"
              name="size"
              onClick={[Function]}
              title="Refine by size: 10"
              value="10"
            >
              10
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="size-12"
              key="16"
              name="size"
              onClick={[Function]}
              title="Refine by size: 12"
              value="12"
            >
              12
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="size-14"
              key="17"
              name="size"
              onClick={[Function]}
              title="Refine by size: 14"
              value="14"
            >
              14
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="size-16"
              key="18"
              name="size"
              onClick={[Function]}
              title="Refine by size: 16"
              value="16"
            >
              16
            </button>
            <button
              className="undrln-btn sample-hover"
              disabled={false}
              id="size-18"
              key="19"
              name="size"
              onClick={[Function]}
              title="Refine by size: 18"
              value="18"
            >
              18
            </button>
          </div>
        </div>
      </FilterSection>
      <FilterSection
        name="price"
        showFilter={true}
        showName="showPrice"
        toggleFilter={[Function]}
        updateFilter={[Function]}
      >
        <div
          className="filter"
          id="filter-price"
        >
          <div
            className="filter-banner"
          >
            <button
              id="filter-price-label"
              name="showPrice"
              onClick={[Function]}
            >
              <h4>
                PRICE
              </h4>
            </button>
            <button
              className="filter-clear-btn"
              disabled={true}
              id="filter-clear-price-btn"
              name="price"
              onClick={[Function]}
              value={null}
            />
            <button
              className="filter-show-btn"
              id="filter-show-price-btn"
              name="showPrice"
              onClick={[Function]}
            >
              <SvgIcon
                name="downArrow"
                width={10}
              >
                <svg
                  className=""
                  height="10px"
                  style={Object {}}
                  viewBox="0 0 451.847 451.847"
                  width="10px"
                  xmlns="http://www.w3.org/2000/svg"
                  xmlnsXlink="http://www.w3.org/1999/xlink"
                >
                  <g>
                    <path
                      d="M225.923,354.706c-8.098,0-16.195-3.092-22.369-9.263L9.27,151.157c-12.359-12.359-12.359-32.397,0-44.751 c12.354-12.354,32.388-12.354,44.748,0l171.905,171.915l171.906-171.909c12.359-12.354,32.391-12.354,44.744,0 c12.365,12.354,12.365,32.392,0,44.751L248.292,345.449C242.115,351.621,234.018,354.706,225.923,354.706z"
                      fill="#6d6c6c"
                      key="0"
                    />
                  </g>
                </svg>
              </SvgIcon>
            </button>
          </div>
          <div
            className="filter-show filter-show-price"
          />
        </div>
      </FilterSection>
      <FilterSection
        list={
          Array [
            "brand1",
            "brand2",
          ]
        }
        name="brand"
        pageQuery={
          Object {
            "department": "accessories",
          }
        }
        showFilter={true}
        showName="showBrand"
        toggleFilter={[Function]}
        updateFilter={[Function]}
      >
        <div
          className="filter"
          id="filter-brand"
        >
          <div
            className="filter-banner"
          >
            <button
              id="filter-brand-label"
              name="showBrand"
              onClick={[Function]}
            >
              <h4>
                BRAND
              </h4>
            </button>
            <button
              className="filter-clear-btn"
              disabled={true}
              id="filter-clear-brand-btn"
              name="brand"
              onClick={[Function]}
              value={null}
            />
            <button
              className="filter-show-btn"
              id="filter-show-brand-btn"
              name="showBrand"
              onClick={[Function]}
            >
              <SvgIcon
                name="downArrow"
                width={10}
              >
                <svg
                  className=""
                  height="10px"
                  style={Object {}}
                  viewBox="0 0 451.847 451.847"
                  width="10px"
                  xmlns="http://www.w3.org/2000/svg"
                  xmlnsXlink="http://www.w3.org/1999/xlink"
                >
                  <g>
                    <path
                      d="M225.923,354.706c-8.098,0-16.195-3.092-22.369-9.263L9.27,151.157c-12.359-12.359-12.359-32.397,0-44.751 c12.354-12.354,32.388-12.354,44.748,0l171.905,171.915l171.906-171.909c12.359-12.354,32.391-12.354,44.744,0 c12.365,12.354,12.365,32.392,0,44.751L248.292,345.449C242.115,351.621,234.018,354.706,225.923,354.706z"
                      fill="#6d6c6c"
                      key="0"
                    />
                  </g>
                </svg>
              </SvgIcon>
            </button>
          </div>
          <div
            className="filter-show filter-show-brand"
          >
            <button
              className="undrln-btn false"
              disabled={false}
              id="brand-brand1"
              key="0"
              name="brand"
              onClick={[Function]}
              title="Refine by brand: Brand1"
              value="brand1"
            >
              Brand1
            </button>
            <button
              className="undrln-btn false"
              disabled={false}
              id="brand-brand2"
              key="1"
              name="brand"
              onClick={[Function]}
              title="Refine by brand: Brand2"
              value="brand2"
            >
              Brand2
            </button>
          </div>
        </div>
      </FilterSection>
    </div>
  </Filter__FilterStyles>
</Filter>
`;
